packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -61,9 +85,14 @@ export class HttpRouteKey {
     if (path !== '/' && (!path.startsWith('/') || path.endsWith('/'))) {
       throw new Error('A route path must always start with a "/" and not end with a "/"');
     }
-    return new HttpRouteKey(`${method ?? HttpMethod.ANY} ${path}`, path);
+    const keyMethod = method ?? HttpMethod.ANY;
+    return new HttpRouteKey(keyMethod, `${keyMethod} ${path}`, path);
```

Comment:
I've now simplified this section so that it's cleaner and easier to avoid mistakes. As far as I can see, HTTP API has only the `$default` and `METHOD /path/here` formats for `key`, so this should work. https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-develop-routes.html

Let me know whether this change works for you.